crates/bevy_reflect/src/serde/de.rs

```diff
@@ -538,6 +560,60 @@ impl<'a, 'de> Visitor<'de> for StructVisitor<'a> {
     }
 }
 
+struct NewtypeStructVisitor<'a> {
+    tuple_struct_info: &'static TupleStructInfo,
+    registry: &'a TypeRegistry,
+    registration: &'a TypeRegistration,
+}
+
+impl<'a, 'de> Visitor<'de> for NewtypeStructVisitor<'a> {
+    type Value = DynamicTupleStruct;
+
+    fn expecting(&self, formatter: &mut Formatter) -> std::fmt::Result {
+        formatter.write_str("reflected newtype struct value")
+    }
+
+    fn visit_newtype_struct<D>(self, deserializer: D) -> Result<Self::Value, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        let mut tuple_struct = DynamicTupleStruct::default();
+
+        let ignored_len = self
+            .registration
+            .data::<SerializationData>()
+            .map(|data| data.len())
+            .unwrap_or(0);
+        let field_len = self
+            .tuple_struct_info
+            .field_len()
+            .saturating_sub(ignored_len);
+
+        if field_len > 1 {
```

Comment:
Actually, I think this should check for exactly 1, rather than 0 or 1. This way, we can safely replace the `self.tuple_struct_info.field_at(0).ok_or_else(...)` with `self.tuple_struct_info.field_at(0).unwrap()`, and also consolidate the error messages.

Reply:
Hm, since this is private, we could also rely on the `TypedReflectDeserializer` to uphold this single-field requirement (since it's doing that already) 🤔

Reply:
Had an idea [here](https://github.com/bevyengine/bevy/pull/8302#discussion_r1157908738)